tests/rule_based_profiler/data_assistant/test_onboarding_data_assistant.py

```diff
@@ -317,6 +319,65 @@ def test_onboarding_data_assistant_result_batch_id_to_batch_identifier_display_n
     )
 
 
+@pytest.mark.big
+def test_onboarding_data_assistant_should_fail_forward(
+    bobby_columnar_table_multi_batch_deterministic_data_context,
+    rule_state_with_domains_and_parameters,
+):
+    """When one rule fails, the rest of the rules should still be executed."""
+    context: DataContext = bobby_columnar_table_multi_batch_deterministic_data_context
+
+    batch_request: dict = {
+        "datasource_name": "taxi_pandas",
+        "data_connector_name": "monthly",
+        "data_asset_name": "my_reports",
+    }
+
+    num_rules = 8  # There are 8 rules in the onboarding data assistant.
+    with mock.patch(
+        "great_expectations.rule_based_profiler.rule.rule.Rule.run",
+    ) as mock_run:
```

Comment:
Do we need to evaluate all 8 rules to test this change? Just thinking about test performance (I know this area is historically quite cumbersome in terms of testing)

Reply:
The rules aren't run at all in this test. They're all mocked out.